### DISCUSSION TITLE
Valhalla roundabout announcements

### DISCUSSION THREAD

**Discussion by Leeb64** [devemux86/cruiser#582] [Q&A] [answered]
I was traveling last week.
On the first day we were in a hurry and I thought: let's give Valhalla a try.
At every(!) roundabout he told us that we should take the "2nd" exit. The direction display also showed a "2".
No matter which exit we had to take...
The correct exit was shown on the map!

**Comment by devemux86** [comment]
@Leeb64 thanks for the report.

Making a test with a random route, I see different roundabout exits in turn instructions screen and navigation.

Please post a route example, so we can check it.

**Reply by Leeb64** [reply]
[Anfahrt .json](https://github.com/user-attachments/files/16051918/Anfahrt.json)

**Reply by devemux86** [reply]
@Leeb64 what profile (vehicle?) and Valhalla options did you use?

**Reply by Leeb64** [reply]
If I haven't changed anything yet:
->motorcycle
Toll: off,
Ferries: off,
highway 50%,
Residential streets 20%,
gravel: off,
Paths:off,
speed 88km/h,
Avoid unpaved!
Fast.

**Reply by Leeb64** [reply]
Here 3 Screenshots. The calculation with Valhalla and the Options from above.
![Screenshot_1](https://github.com/devemux86/cruiser/assets/120722053/028dd92a-f3cd-4c45-8cd0-25df294544b4)
![Screenshot_2](https://github.com/devemux86/cruiser/assets/120722053/ce8d9868-096e-4b53-87fa-6da641c3fffb)
![Screenshot_3](https://github.com/devemux86/cruiser/assets/120722053/89ea5f1b-8bfa-4b02-80a1-fe4dd328e718)
No matter which exit, it is always announced and displayed:  Take the 2. exit.
Even with new tests the same error occurs.

**Reply by devemux86** [reply]
@Leeb64 

Navigation uses the turn instructions generated by the routing services.
So this is about route and instruction calculation by the routing services.

As mentioned above, I made a random Valhalla route, it had different exits, so they were announced correctly.

The Valhalla turn directions in your example all show the 2nd exit, so obviously those are announced.
Are these Valhalla turn instructions wrong?

Below I tried to make the same route on Valhalla website. All roundabout exits are "2":
https://github.com/devemux86/cruiser/discussions/582#discussioncomment-9924631

Can you post a link with the route from [Valhalla website](https://valhalla.openstreetmap.de/) so we can compare the turn instructions?

**Reply by Leeb64** [reply]
Most (80%) turn-by-turn directions are correct.
This is because the second exit means “straight ahead”.
But the 2nd exit is always announced - even though something different (correct) is shown on the map.
The turn-by-turn instructions in roundabouts (with Valhalla) do not work properly.

**Reply by devemux86** [reply]
@Leeb64

We need to separate 2 things, as there is still confusion about the report:

- If [Valhalla website](https://valhalla.openstreetmap.de/) shows the same turn instructions, then Cruiser reads them correctly.

- If Valhalla turn instructions are not correct, then as with all routers, it is best to report to them so they can check.

---

Also what do you mean "the second exit means “straight ahead”'? Either it is a roundabout with exits or not.

**Reply by Leeb64** [reply]
The second exit at a roundabout is usually the exit opposite the entrance (4x roundabout).
Basicly straight ahead!

**Comment by devemux86** [comment, marked as answer]
This should be the route on [Valhalla website](https://valhalla.openstreetmap.de/directions?profile=motorcycle&wps=8.04576873779297%2C49.206439193053086%2C6.722087860107423%2C49.10815219044901%2C3.5854911804199223%2C47.8116558652381%2C3.1891250610351567%2C47.623520868939536) with motorcycle and default options.

You can check the maneuvers in the website. Is there anything different in Cruiser?
